packages/shortest/src/ai/client-provider.ts

```diff
@@ -0,0 +1,140 @@
+import assert from "node:assert";
+import { createAnthropic } from "@ai-sdk/anthropic";
+import { createAmazonBedrock } from "@ai-sdk/amazon-bedrock";
+import { LanguageModelV1 } from "ai";
+import { LLMConfig } from "../types/ai";
+
+/**
+ * Creates an AI client based on the provided configuration.
+ *
+ * @param config - The configuration for the language model.
+ * @returns A LanguageModelV1 instance.
+ * @throws Will throw an error if the provider or model is unsupported or if required environment variables are missing.
+ */
+export function createAIClient(config: LLMConfig): LanguageModelV1 {
+  const { provider, region, accessKeyId, secretAccessKey } = config;
+  const model = resolveWhitelistedProviderModel(provider, config.model);
+
+  switch (provider) {
+    case "anthropic":
+      return createAnthropicClient(model);
+
+    case "amazon-bedrock":
+      return createAmazonBedrockClient({
+        region,
+        accessKeyId,
+        secretAccessKey,
+        model,
+      });
+
+    default:
+      throw new Error(`Unknown provider: ${provider}`);
+  }
+}
+
+/**
+ * Resolves and creates the Anthropic client.
+ *
+ * @param model - The provider-specific model string.
+ * @returns A LanguageModelV1 instance for Anthropic.
+ * @throws Will throw an error if the API key is missing.
+ */
+function createAnthropicClient(model: string): LanguageModelV1 {
+  const apiKey = process.env.ANTHROPIC_API_KEY;
```

Comment:
How would this work if the `apiKey` is coming from the config?

Generally, I think `parseConfig` (or something similar) should:
* parse the original config data
* apply ENVs as default if needed
* validate config

That way, `globalConfig` becomes the source of truth for all config-related stuff.